content/docs/concepts/remote_write_spec_2_0.md

```diff
@@ -0,0 +1,367 @@
+
+* [`prometheus.WriteRequest`](./remote_write_spec.md#protocol) introduced in the Remote Write 1.0 specification. As of 2.0 the `prometheus.WriteRequest` message is deprecated. It SHOULD be used only for compatibility reasons. Sender and Receiver MAY NOT support `prometheus.WriteRequest`.
+* `io.prometheus.write.v2.Request` introduced in this specification and defined [below](#ioprometheuswritev2request-proto-schema). Senders and Receivers SHOULD use `io.prometheus.write.v2.Request` when possible. Sender and Receiver MUST support `io.prometheus.write.v2.Request`.
+
+The encoded message MUST be compressed with [Google’s Snappy](https://github.com/google/snappy). The block format MUST be used -- the framed format MUST NOT be used.
+
+Sender MUST send encoded and compressed proto message in the body of an HTTP POST request and send it to the Receiver via HTTP at a provided URL path. The Receiver MAY specify any HTTP URL path to receive metrics.
+
+Sender MUST send the following reserved headers with the HTTP request:
+
+* `Content-Encoding: <compression>`
+
+  Content encoding request header MUST follow [the RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-encoding). Sender MUST use the `snappy` value. Receiver MUST support `snappy` compression. New, optional compression algorithms might come in 2.x or beyond.
+
+* `Content-Type: application/x-protobuf` or `Content-Type: application/x-protobuf;proto=<fully qualified name>`
+
+  Content type request header MUST follow [the RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-type). Sender MUST use `application/x-protobuf` as the only media type. Sender MAY add `;proto=` parameter to the header's value to indicate the fully qualified name of the protobuf message (schema) that was used, from the two mentioned above. As a result, Sender MUST send any of the three supported header values:
+
+  For the deprecated message introduced in PRW 1.0, identified by `prometheus.WriteRequest`:
+    * `Content-Type: application/x-protobuf`
+    * `Content-Type: application/x-protobuf;proto=prometheus.WriteRequest`
+  For the message introduced in PRW 2.0, identified by `io.prometheus.write.v2.Request`:
+    * `Content-Type: application/x-protobuf;proto=io.prometheus.write.v2.Request`
+  
+  When talking to 1.x Receiver, the Sender SHOULD use `Content-Type: application/x-protobuf` for backward compatibility. Otherwise, Sender SHOULD use `Content-Type: application/x-protobuf;proto=io.prometheus.write.v2.Request`. More proto messages might come in 2.x or beyond.
+
+* `User-Agent: <name & version of the sender>`
+* `X-Prometheus-Remote-Write-Version: <remote write spec major and minor version>`
+
+  When talking to 1.x Receiver, the Sender MUST use `X-Prometheus-Remote-Write-Version: 0.1.0` for backward compatibility. Otherwise, Sender SHOULD use the newest remote write version it is compatible with e.g. `X-Prometheus-Remote-Write-Version: 2.0.0`.
```

Comment:
did we not allow for `X-Prometheus-Remote-Write-Version: 1.0` in the 1.0 spec as well?

Reply:
nevermind, confirmed myself that the 1.0 spec says `0.1.0` must be used